action.php

```diff
@@ -70,22 +70,25 @@
         $OUTPUT = $PAGE->get_renderer('tool_mfa');
         $form = new setup_factor_form($currenturl, array('factorname' => $factor));
 
-        if ($form->is_cancelled()) {
-            redirect($returnurl);
-        }
-
         if ($form->is_submitted()) {
-            if ($data = $form->get_data()) {
-                if ($factorobject && $factorobject->setup_user_factor($data)) {
-                    $event = \tool_mfa\event\user_setup_factor::user_setup_factor_event($USER, $factorobject->get_display_name());
-                    $event->trigger();
-
-                    redirect($returnurl);
-                } else {
-                    print_error('error:setupfactor', 'tool_mfa', $returnurl);
+            $form->is_validated();
+
+            if ($form->is_cancelled()) {
+                redirect($returnurl);
+            } else {
```

Comment:
a tiny thing, do you need this extra nested else? less nesting the better. redirect() will exit on the line above

Reply:
You're right. Let me fix it, plz